pkg/block/azure/adapter.go

```diff
@@ -423,12 +425,26 @@ func (a *Adapter) GetProperties(ctx context.Context, obj block.ObjectPointer) (b
 	if err != nil {
 		return block.Properties{}, err
 	}
+	etag := calcETag(props.ContentMD5, props.ETag)
 	return block.Properties{
 		StorageClass: props.AccessTier,
 		LastModified: apiutil.Value(props.LastModified),
+		ETag:         etag,
 	}, nil
 }
 
+// calcETag etag set by content md5 with fallback to use etag value
+func calcETag(contentMD5 []byte, etag *azcore.ETag) string {
+	if contentMD5 != nil {
+		return hex.EncodeToString(contentMD5)
+	}
```

Comment:
Please add a comment explaining why it makes sense _sometimes_ to take MD5 and other times to take the ETag.  Is this SOP on Azure?